server/src/main/java/org/candlepin/model/User.java

```diff
@@ -254,32 +243,24 @@ public void removeRole(Role r) {
     }
 
     /**
-     * Full list of permissions for this user.
-     *
-     * Includes those from roles stored in the database, as well as those explicitly added
-     * by the user service adapter.
-     *
-     * @return full list of permissions for this user.
+     * Clears any existing roles for this user.
      */
-    @XmlTransient
-    public Set<Permission> getPermissions() {
-        PermissionFactory permFactory = new PermissionFactory();
-        Set<Permission> perms = new HashSet<>();
-        for (Role r : getRoles()) {
-            perms.addAll(permFactory.createPermissions(this, r.getPermissions()));
-        }
-        perms.addAll(this.permissions);
-        return perms;
-    }
+    public void clearRoles() {
+        Set<Role> cleared = this.roles;
+
+        if (cleared != null) {
+            this.roles = new HashSet<>();
 
-    public void addPermissions(Permission permission) {
-        this.permissions.add(permission);
+            for (Role role : cleared) {
+                role.removeUser(this);
+            }
+        }
     }
 
     /**
      * @return if the user has the SUPER_ADMIN role
      */
-    public boolean isSuperAdmin() {
+    public Boolean isSuperAdmin() {
```

Comment:
Unless "isSuperAdmin" returning null is meaningful, I vote that we do not change this.

Reply:
This is a bit of an odd one. If we want to make life easy on things going to and from the adapter layer, we need to implement UserInfo. While we could make UserInfo's definition kick out a primitive, our UserDTO would then need extra work to get around it. 

Also, we need some sentinel value to represent "no change," which is currently null in most other cases.